src/ExpressionEvaluator/Core/Source/ResultProvider/ResultProvider.cs

```diff
@@ -679,5 +689,43 @@ internal static void ContinueWithExceptionHandling(CompletionRoutine onCompleted
                 onException(e);
             }
         }
+
+        private delegate void WorkListCompletionRoutine(WorkList workList);
+
+        private sealed class WorkList
+        {
+            internal readonly DkmWorkList InnerWorkList;
+            private readonly CompletionRoutine<Exception> _onException;
+            private WorkListCompletionRoutine _completionRoutine;
```

Comment:
As far as I can tell, there are three kinds of completion routines floating around, WorkListCompletionRoutine, CompletionRoutine<T>, and DkmCompletionRoutine<T>.  However, we seem to call locals of all three types "completionRoutine".  I think the code would be much easier to follow if we gave them distinct names.